src/graph_impl/mod.rs

```diff
@@ -1493,6 +1493,31 @@ where
         g
     }
 
+    /// Create a new `Graph` by mapping node and edge weights to new values,
+    /// consuming the current graph.
+    ///
+    /// The resulting graph has the same structure and the same graph indices
+    /// as `self`.
+    ///
+    /// If you want a non-destructive version of this function, see `map`.
```

Comment:
```suggestion
    /// If you want a non-consuming version of this function, see `map`.
```
I think, I would prefer to write non-consuming instead of destructive, since destruction might imply more than just giving up ownership of the value.